clients/src/main/java/org/apache/kafka/clients/admin/DeleteRecordsResult.java

```diff
@@ -33,7 +33,7 @@
 
     private final Map<TopicPartition, KafkaFuture<DeletedRecords>> futures;
 
-    DeleteRecordsResult(Map<TopicPartition, KafkaFuture<DeletedRecords>> futures) {
+    public DeleteRecordsResult(Map<TopicPartition, KafkaFuture<DeletedRecords>> futures) {
```

Comment:
Good point. I will just keep the KafkaFuture from the result in `TaskManager` then.

Reply:
@dguy Tried to remove this constructor from `TaskManagerTest` but it turns out ugly. So I plan to use the `MockAdminClient` after #4242 instead of the mock and create the object.